src/instructlab/data/generate.py

```diff
@@ -210,6 +220,16 @@ def generate(
         except Exception as exc:
             click.secho(f"Failed to start server: {exc}", fg="red")
             raise click.exceptions.Exit(1)
+
+        # If batch size is not set explicitly, disable batching when running
+        # with the local llama.cpp server
+        if isinstance(backend_instance, llama_cpp_server) and batch_size is None:
```

Comment:
Should this be:

```
if llama-cpp
  if batch_size is None
    batch_size=0
  else
    logger.warning(...)
```

Reply:
Kind of - but I think we want to set `batch_size=0` in all cases for llama-cpp, even if the user specifies a batch size